message Track {
	optional string id = 2; //i dont know what this is
	optional int32 creation = 3; //unix timestamp seconds - only a guess
	optional int32 lastPlayed = 4; //unix timestamp seconds - only a guess
	optional string title = 6; //fairly certain
	optional string artist = 7; //fairly certain
	optional string composer = 8; 
	optional string album = 9;
	optional string albumArtist = 10; 
	optional int32 year = 11; 
	optional string comment = 12;
	optional int32 track = 13; 
	optional string genre = 14;	//fairly certain
	optional int32 duration = 15; //fairly certain
	optional int32 beatsPerMinute = 16; 
	optional int32 playCount = 20;
	optional int32 totalTracks = 26; 
	optional int32 disc = 27;
	optional int32 totalDiscs = 28;
	optional int32 u11 = 31; //seems to be exclusively 1
	optional int32 fileSize = 32; //file size
	optional int32 u13 = 37; //seems to be exclusively 0
	optional int32 u14 = 38; //seems to be exclusively 0
	optional int32 bitrate = 44; //mostly certain
	optional string u15 = 53; //seems to be exclusively "-383260437"
	optional int32 u16 = 61; 
}


message MetadataRequest {
	repeated Track tracks = 1;
	optional string address = 2; //I don't know what this is
}

message QueuedUpload {
	optional string id = 1;
	optional int32 u0 = 2; //i dont know but this is always 2
	optional string serverId = 3;  //That's what this number is referred to later as during the uploadsj/rupio
}

message Status {
	optional int32 u0 = 1;
	optional int32 u1 = 2;
	optional int32 u2 = 3; //like totally always five
	optional int32 u3 = 4; //this is like totally always 6000
	optional int32 u4 = 5; 
	optional int32 u5 = 6; 
}


message TrackResponse {
	repeated string ids = 2;
	repeated QueuedUpload uploads = 3;	
}

message MetadataResponse {
	optional int32 u0 = 1; //no idea what this is, but the value is 1
	optional TrackResponse response = 2;
	optional Status state = 6;
}

message UploadAuth {
	optional string address = 1;
	optional string hostname = 2;
}


message Quota {
	optional int32 maximumTracks = 1;
	optional int32 availableTracks = 2;
	optional int32 totalTracks = 3;
}

message ClientState {
	optional string address = 1;
}

message ClientStateResponse {
	optional int32 u0 = 1;
	optional Status status = 6;
	optional Quota quota = 8;
}

message UploadAuthResponse {
	optional int32 u0 = 1;
	optional Status status = 6;
	optional int32 u1 = 11;
	optional int32 u2 = 12;
}